app/services/discourse_reactions/reaction_manager.rb

```diff
@@ -7,27 +7,31 @@ def initialize(reaction_value:, user:, guardian:, post:)
       @user = user
       @guardian = guardian
       @post = post
+      @like = @post.post_actions.find_by(user: @user, post_action_type_id: post_action_like_type)
     end
 
     def toggle!
       ActiveRecord::Base.transaction do
+        return if (@like && !@guardian.can_delete_post_action?(@like)) || (is_reacted_by_user && !@guardian.can_delete_reaction_user?(old_reacted_user))
+        @reaction = reaction_scope&.first_or_create
+        @reaction_user = reaction_user_scope
         @reaction_value == DiscourseReactions::Reaction.main_reaction_id ? toggle_like : toggle_reaction
       end
     end
 
     private
 
     def toggle_like
-      like = @post.post_actions.find_by(user: @user, post_action_type_id: post_action_like_type)
-      raise Discourse::InvalidAccess if like && !@guardian.can_delete_post_action?(like)
-      like ? remove_shadow_like : add_shadow_like
+      remove_shadow_like if @like
```

Comment:
```ruby
if @like
  remove_shadow_like
else
  add_shadow_like
end
```

Reply:
it is the same as the case of the reaction, if the user wants to change its like, we need to first remove the remove and then add a new one